src/Components/Movies/Movies.tsx

```diff
@@ -4,39 +4,39 @@ import Pagination from "@mui/material/Pagination";
 import CircularProgress from "@mui/material/CircularProgress";
 import { BsStarFill } from "react-icons/bs";
 
-
-import { getMovies } from "../../Api/Movies/MoviesApi";
 import "./Movies.css";
 import { useAppSelector } from "../../App/hooks";
-import { posterMoviesCount } from "../../Api/Dashboard/DashboardApi";
+import { fetchMovieGenres, fetchMovies } from "../../Api/TMDB/tmdbService";
 
 function Movies() {
   const { search } = useLocation();
   const navigate = useNavigate();
   const ThemeMenu = useAppSelector(state => state.ThemeMenu);
   const [page, setPage] = useState(Number(search.split("=")[1]));
   const [moviesList, setMoviesList] = useState([]);
+  const [genresList, setGenresList] = useState([]);
   const [moviesState, setMoviesState] = useState("INITIAL");
-  const [posterCount, setPosterCount] = useState(100);
 
   useEffect(() => {
-    fetchPosterCount();
-    setPage(Number(search.split("=")[1]));
-    fetchMovies(Number(search.split("=")[1]));
-  }, []);
+    getGenres()
+  }, [search]);
 
-  const fetchPosterCount = async () => {
-    const { status, data } = await posterMoviesCount();
+  const getGenres = async () => {
+    setMoviesState("LOADING");
```

Comment:
Better to keep these states as enums. You shouldn't have to hard code them every time.

Reply:
Created an enum "apiStatus" for api state handling